### PR TITLE
Docs, part 16

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #358

### CODE REVIEW

File: src/Rule/Nested.php [hunk 1]
```diff
@@ -62,32 +113,71 @@ final class Nested implements
     private iterable|null $rules;
```
Comment (arogachev) [inline src/Rule/Nested.php:113]: Add description.

File: src/Rule/Nested.php [hunk 2]
```diff
@@ -62,32 +113,71 @@ final class Nested implements
...
 
     /**
-     * @param iterable|object|string|null $rules Rules for validate value that can be described by:
+     * @param iterable|object|string|null $rules
```
Comment (arogachev) [inline src/Rule/Nested.php:116]: Add description.

**Review by vjik — APPROVED** [review]

File: src/Rule/Nested.php [hunk 3]
```diff
@@ -36,9 +37,86 @@
...
+ *
+ * ```php
+ * $rule = new Nested([
+ *     'title' => [new HasLength(max: 255)],
+ *     'author.name' => [new HasLength(min: 1)],
+ *     'files.*.url' => [new Url()],
+ * ]);
+ * ```
+ *
+ * Note that the maximum nesting level is 2, a deeper one requires wrapping with another `Nested` instance or using
```
Comment (samdark) [inline src/Rule/Nested.php:73]: Doc is fine but the limitation itself isn't great. Is there an issue to improve it?

File: src/Rule/Nested.php [hunk 4]
```diff
@@ -36,9 +37,86 @@
...
+ *     ]),
+ * ];
+ * ```
+ *
+ * For more examples please refer to the guide.
+ *
+ * It's also possible to use DTO objects with PHP attributes, see {@see ObjectDataSet} documentation and guide for
+ * details.
+ *
+ * Supports propagation of options (see {@see PropagateOptionsHelper::propagate()} for supported options and
```
Comment (samdark) [inline src/Rule/Nested.php:113]: This sentence doesn't look alright. Is it possible to make it better?

Comment (arogachev) [reply src/Rule/Nested.php:113]: Sorry, but could you tell what exactly is wrong in your opinion?

File: src/Rule/Nested.php (hunk 3 above)
Comment (arogachev) [reply src/Rule/Nested.php:73]: I was going to create one. :ok_hand:

Comment (arogachev) [reply src/Rule/Nested.php:73]: #508

**Pull request merged**